plugin-client/src/main/java/com/cloudera/nav/plugin/client/NavApiCient.java

```diff
@@ -71,18 +78,105 @@ public void registerModels(Collection<Object> models) {
+    UpdatedResults updatedResults;
+    Map<String, Integer> currentMarker = getCurrentMarker();
+    try {
+      String currentMarkerRep = new ObjectMapper().writeValueAsString(currentMarker);
+      String queryString = "";
+      updatedResults = aggUpdatedResults(currentMarkerRep, queryString);
+      return updatedResults;
+    } catch (IOException e){
+      System.err.println(e.getMessage());
+      throw Throwables.propagate(e);
+    }
+  }
+
+  /**Returns all of the entities and relations in the database that have been updated or added since the source iterations indicated by the marker
+   *
+   * @param markerRep JSON representation of sourceId : sourcesourceExtractIteration
+   * @return
+   */
+
+  public UpdatedResults getAllUpdated(String markerRep){
+    UpdatedResults updatedResults;
+    Map<String, Integer> currentMarker = getCurrentMarker();
+    try {
+      String currentMarkerRep = new ObjectMapper().writeValueAsString(currentMarker);
+      Map<String, Integer> marker = new ObjectMapper().readValue(markerRep, new TypeReference<Map<String, Integer>>(){});
+      String extractorQueryString = getExtractorQueryString(marker, currentMarker);
+      updatedResults = aggUpdatedResults(currentMarkerRep, extractorQueryString);
+      return updatedResults;
+    } catch (IOException e) {
+      System.err.println(e.getMessage());
+      throw Throwables.propagate(e);
+    }
+  }
+
+  public UpdatedResults aggUpdatedResults(String markerRep, String queryString){
+    UpdatedResults updatedResults;
+    Map<String, Integer> currentMarker = getCurrentMarker();
+    ParameterizedTypeReference<Map<String, Object>[]> mapRef = new ParameterizedTypeReference<Map<String, Object>[]>(){};
+    Map<String, Object>[] entityResult = navResponse("entities", queryString, mapRef);
+    System.out.println("ENTITIES: \n"+ entityResult.toString());
+    Map<String, Object>[] relationsResult = navResponse("relations", queryString, mapRef);
+    UpdatedResults res = new UpdatedResults(markerRep, Arrays.asList(entityResult), Arrays.asList(relationsResult));
+    return res;
+  }
+
+  /**
+   *
+   * @param type
+   * @param extractorQuery
+   * @param resultClass
+   * @param <T>
+   * @return
+   */
+  public <T> T navResponse(String type, String extractorQuery, ParameterizedTypeReference<T> resultClass){
+    RestTemplate restTemplate = new RestTemplate();
+    HttpHeaders headers = getAuthHeaders();
+    HttpEntity<String> request =new HttpEntity<String>(headers);
+    String url = getUrl(type, extractorQuery);;
+    ResponseEntity<T> response = restTemplate.exchange(url, HttpMethod.GET, request, resultClass);
+    T responseResult = response.getBody();
+    return responseResult;
+  }
+
+  /**
+   *
+   * @return
+   */
+  public Map<String, Integer> getCurrentMarker(){
+    Collection<Source> sources = getAllSources(); // Replace with sourceCacheByUrl and loadAllSources ?
+    HashMap<String, Integer> newMarker = new HashMap<>();
```

Comment:
Use interfaces where you can, so declare this as type Map. Also prefer Maps.newHashMap, Sets.newHashSet, List.newArrayList, etc to directly calling the constructor here.